S/solar/build_tarballs.jl

```diff
@@ -0,0 +1,44 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "solar"
+version = v"0.3.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/bbopt/solar.git", "d0c2932a3f25d4b0a71010b87b1f929c6fc1e020"),
+    DirectorySource("./bundled")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/solar/src
+if [[ "${target}" == *mingw* ]]; then
+    atomic_patch -p1 "${WORKSPACE}/srcdir/patches/windows.patch"
+fi
+make
+mkdir -p $bindir
+if [[ "${target}" == *mingw* ]]; then
+    cp ../bin/solar.exe $bindir/solar.exe
+else
+  cp ../bin/solar $bindir/solar
+end
+exit
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = expand_cxxstring_abis(supported_platforms())
+
+
+# The products that we will ensure are always built
+products = [
+    ExecutableProduct("solar", :solar)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"9.1.0")
```

Comment:
Since this is a new recipe, would you be ok to require Julia v1.6?  That would allow us to build for new platforms, like aarch64-apple-darwin (aka Apple Silicon)

Reply:
Yes, let's build on the new platforms.